tests/runtest.sh

```diff
@@ -311,8 +311,25 @@ function create_core_overlay {
         rm -f -r "$coreOverlayDir"
     fi
     mkdir "$coreOverlayDir"
-    (cd $coreFxBinDir && find . -iname '*.dll' \! -iwholename '*test*' \! -iwholename '*/ToolRuntime/*' -exec cp -f -u '{}' "$coreOverlayDir/" \;)
-    cp -f "$coreFxNativeBinDir/Native/"*.so "$coreOverlayDir/" 2>/dev/null
+
+    local OSName=$(uname -s)
+    case $OSName in
+        Linux)
+            (cd $coreFxBinDir && find . -iname '*.dll' \! -iwholename '*test*' \! -iwholename '*/ToolRuntime/*' -exec cp -f -u '{}' "$coreOverlayDir/" \;)
+            cp -f "$coreFxNativeBinDir/Native/"*.so "$coreOverlayDir/" 2>/dev/null
+            ;;
+
+        Darwin)
+            cp -f "$coreFxNativeBinDir/Native/"*.dylib "$coreOverlayDir/" 2>/dev/null
+            (cd $coreFxBinDir && find . -iname '*.dll' \! -iwholename '*test*' \! -iwholename '*/ToolRuntime/*' -exec cp -f '{}' "$coreOverlayDir/" \;)
+            ;;
+        *)
+            echo "Unsupported OS $OSName detected, configuring as if for Linux"
+            (cd $coreFxBinDir && find . -iname '*.dll' \! -iwholename '*test*' \! -iwholename '*/ToolRuntime/*' -exec cp -f -u '{}' "$coreOverlayDir/" \;)
+            cp -f "$coreFxNativeBinDir/Native/"*.so "$coreOverlayDir/" 2>/dev/null
+            ;;
+    esac
```

Comment:
Have you considered using rsync rather than cp so that it's unified and we don't need to special case the `cp` args for OS X? If you do that and make 'dylib'/'so' a variable like `libraryExtension` or something this can be much cleaner.